### ISSUE TITLE
Implement a proper margin convention

### ISSUE THREAD

**Issue by ivanvanderbyl** [ivanvanderbyl/maximum-plaid#25]
http://bl.ocks.org/mbostock/3019563

**Comment by xomaczar** [comment]
I am also interested in responsive/dynamic plot dimensions. I've noticed plaid-plot extends Coordinates mixin which provides support for resizing the plot. However, the area helper only supports fixed height/width - which are used by various scales.
I've implemented the plot-dimension component which yields the dynamic height/width to the area helper (using the resize mixin). I'd like to get your opinion on the best way to provide support for responsive plot dimensions.

**Comment by xomaczar** [comment]
@ivanvanderbyl ping

**Comment by ivanvanderbyl** [comment]
Hey @xomaczar sorry been flat out with real work all week. So what I was thinking here for resizing was implementing a keyword, like `auto` for width or height which will use the width of the parent, as well as respond to resize events. I'd also like to implement the margin convention from above.

**Comment by xomaczar** [comment]
Yeah that what I was thinking - we need to add another component that provides its height/width to the area helper in order to keep plot's innerHeight responsive. I'll try to make a PR with an initial implementation and we can go from there.

**Comment by xomaczar** [comment]
Just heard your podcast with Taras - are you thinking about pivoting this library.
